bundles/specmate-connectors-api/src/com/specmate/connectors/api/IRequirementsSource.java

```
@@ -23,6 +25,20 @@
 	 */
 	IContainer getContainerForRequirement(Requirement requirement) throws SpecmateException;
 
-	boolean authenticate(String username, String password) throws SpecmateException;
+	/**
+	 * Returns a set of projects given credentials have access to. Returns null if

```

wouldn't it be better to throw an exception or return an empty list if the credentials are invalid instead of returning null?

---

Good point. However, I do have issues throwing exceptions for expectable cases (such as wrong credentials, which are no exceptions). An empty set is by far better than null.

I updated the documentation and removed null-check